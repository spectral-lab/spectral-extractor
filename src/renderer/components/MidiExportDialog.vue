<template>
  <v-dialog
    v-model="showDialog"
    width="500"
  >
    <v-card>
      <v-card-title
        class="headline accent darken-2"
        primary-title
      >
        MIDI Export
      </v-card-title>

      <v-card-text>
        <v-radio-group
          v-model="target"
          style="margin-top: 0"
          mandatory
        >
          <v-radio
            :value="ALL_TRACKS"
            label="All Tracks"
          />
          <v-radio
            :value="SELECTED_TRACKS"
            label="Selected Tracks"
          />
          <v-radio
            :value="ALL_CLIPS"
            label="All Clips"
          />
          <v-radio
            :value="SELECTED_CLIPS"
            label="Selected Clips"
          />
        </v-radio-group>
      </v-card-text>

      <v-divider />

      <v-card-actions>
        <v-spacer />
        <v-btn
          @click="handleClick"
          color="accent darken-1"
        >
          Export
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-dialog>
</template>

<script>
import { ALL_CLIPS, ALL_TRACKS, SELECTED_TRACKS, SELECTED_CLIPS } from '../constants/midi-export-options';

export default {
  data () {
    return {
      showDialog: true,
      target: ALL_TRACKS,
      ALL_CLIPS,
      ALL_TRACKS,
      SELECTED_CLIPS,
      SELECTED_TRACKS
    };
  },
  methods: {
    show () {
      if (this.showDialog === false) this.showDialog = true;
    },
    hide () {
      if (this.showDialog === true) this.showDialog = false;
    },
    handleClick () {
      this.hide();
      this.$emit('click', this.target);
    }
  }
};
</script>

<style scoped>

</style>
