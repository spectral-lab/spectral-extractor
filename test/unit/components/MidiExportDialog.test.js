import { shallowMount } from '@vue/test-utils';
import MidiExportDialog from '../../../src/renderer/components/dialogs/MidiExportDialog';
import Vue from 'vue';
import vuetify from 'vuetify';

Vue.use(vuetify);

describe('Component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(MidiExportDialog);
  });
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
