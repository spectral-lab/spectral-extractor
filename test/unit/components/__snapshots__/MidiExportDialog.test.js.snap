// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<v-dialog-stub
  contentclass=""
  maxwidth="none"
  origin="center center"
  transition="dialog-transition"
  width="500"
>
  <v-card-stub
    tag="div"
  >
    <v-card-title-stub
      class="headline accent darken-2"
      primarytitle="true"
    >
      
      MIDI Export
    
    </v-card-title-stub>
     
    <v-card-text-stub>
      <v-radio-group-stub
        backgroundcolor=""
        column="true"
        errorcount="1"
        errormessages=""
        height="auto"
        mandatory="true"
        messages=""
        rules=""
        style="margin-top: 0px;"
        successmessages=""
        value="ALL_TRACKS"
        valuecomparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date && b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}"
      >
        <v-radio-stub
          color="accent"
          label="All Tracks"
          officon="$vuetify.icons.radioOff"
          onicon="$vuetify.icons.radioOn"
          ripple="true"
          value="ALL_TRACKS"
        />
         
        <v-radio-stub
          color="accent"
          label="Selected Tracks"
          officon="$vuetify.icons.radioOff"
          onicon="$vuetify.icons.radioOn"
          ripple="true"
          value="SELECTED_TRACKS"
        />
         
        <v-radio-stub
          color="accent"
          label="All Clips"
          officon="$vuetify.icons.radioOff"
          onicon="$vuetify.icons.radioOn"
          ripple="true"
          value="ALL_CLIPS"
        />
         
        <v-radio-stub
          color="accent"
          label="Selected Clips"
          officon="$vuetify.icons.radioOff"
          onicon="$vuetify.icons.radioOn"
          ripple="true"
          value="SELECTED_CLIPS"
        />
      </v-radio-group-stub>
    </v-card-text-stub>
     
    <v-divider-stub />
     
    <v-card-actions-stub>
      <v-spacer-stub />
       
      <v-btn-stub
        activeclass="v-btn--active"
        color="accent darken-1"
        tag="button"
        type="button"
      >
        
        Export
      
      </v-btn-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-dialog-stub>
`;
